desc/integrals/_bounce_utils.py

```diff
@@ -700,127 +845,315 @@ def interp_to_argmin_hard(h, points, knots, g, dg_dz, method="cubic"):
-    include_knots=False,
-    knot_transparency=0.2,
-    include_legend=True,
+def interp_fft_to_argmin(
+    NFP, T, h, points, knots, g, dg_dz, beta=-100, upper_sentinel=1e2
 ):
-    """Plot the piecewise polynomial ``ppoly``.
+    """Interpolate ``h`` to the deepest point of ``g`` between ``z1`` and ``z2``.
+
+    Let E = {ζ ∣ ζ₁ < ζ < ζ₂} and A = argmin_E g(ζ). Returns mean_A h(ζ).
 
     Parameters
     ----------
-    ppoly : PPoly
-        Piecewise polynomial f.
-    num : int
-        Number of points to evaluate for plot.
-    z1 : jnp.ndarray
-        Shape (k.shape[0], W).
-        Optional, intersects with ∂f/∂z <= 0.
-    z2 : jnp.ndarray
-        Shape (k.shape[0], W).
-        Optional, intersects with ∂f/∂z >= 0.
-    k : jnp.ndarray
-        Shape (k.shape[0], ).
-        Optional, k such that f(z) = k.
-    k_transparency : float
-        Transparency of intersect lines.
-    klabel : str
-        Label of intersect lines.
-    title : str
-        Plot title.
-    hlabel : str
-        Horizontal axis label.
-    vlabel : str
-        Vertical axis label.
-    show : bool
-        Whether to show the plot. Default is true.
-    start : float
-        Minimum z on plot.
-    stop : float
-        Maximum z on plot.
-    include_knots : bool
-        Whether to plot vertical lines at the knots.
-    knot_transparency : float
-        Transparency of knot lines.
-    include_legend : bool
-        Whether to include the legend in the plot. Default is true.
+    T : PiecewiseChebyshevSeries
+        Set of 1D Chebyshev spectral coefficients of θ along field line.
+        {θ_α : ζ ↦ θ(α, ζ) | α ∈ A} where A = (α₀, α₁, …, αₘ₋₁) is the same
+        field line.
+    h : jnp.ndarray
+        Shape (..., grid.num_theta, grid.num_zeta)
+        Periodic function evaluated on tensor-product grid in (ρ, θ, ζ) with
+        uniformly spaced nodes (θ, ζ) ∈ [0, 2π) × [0, 2π/NFP).
+        Preferably power of 2 for ``grid.num_theta`` and ``grid.num_zeta``.
+    points : jnp.ndarray
+        Shape (..., num well).
+        Boundaries to detect argmin between.
+        First (second) element stores left (right) boundaries.
+    knots : jnp.ndarray
+        Shape (knots.size, ).
+        z coordinates of spline knots. Must be strictly increasing.
+    g : jnp.ndarray
+        Shape (..., knots.size - 1, g.shape[-1]).
+        Polynomial coefficients of the spline of g in local power basis.
+        Last axis enumerates the coefficients of power series. Second to
+        last axis enumerates the polynomials that compose a particular spline.
+    dg_dz : jnp.ndarray
+        Shape (..., knots.size - 1, g.shape[-1] - 1).
+        Polynomial coefficients of the spline of ∂g/∂z in local power basis.
+        Last axis enumerates the coefficients of power series. Second to
+        last axis enumerates the polynomials that compose a particular spline.
+    beta : float
+        More negative gives exponentially better approximation at the
```

Comment:
should specify this is a factor multiplying the points for the softargmax

Reply:
in that case it'd be better to just specify the formula.

if you want please add the unicode https://www.unicodeit.net/ version of 
![Screenshot from 2024-10-23 12-09-20](https://github.com/user-attachments/assets/de5da6af-cced-4489-bbbc-c8709a739103)

Reply:
unicode doesn't love the formula. That's fine.